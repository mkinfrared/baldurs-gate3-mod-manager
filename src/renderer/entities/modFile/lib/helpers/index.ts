export { extractPakFile } from "./extractPakFile";

export { getModFiles } from "./getModFiles";

export { getModInfo } from "./getModInfo";

export { readZipInfo } from "./readZipInfo";
