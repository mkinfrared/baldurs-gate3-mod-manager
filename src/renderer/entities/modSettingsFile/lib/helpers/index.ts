export { activateMod } from "./activateMod";

export { deactivateMod } from "./deactivateMod";

export { getCurrentSettings } from "./getCurrentSettings";

export { getInstalledMods } from "./getInstalledMods";

export { getModInfo } from "./getModInfo";

export { getNodeValue } from "./getNodeValue";
