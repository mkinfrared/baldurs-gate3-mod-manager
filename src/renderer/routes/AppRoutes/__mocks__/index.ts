export { AppRoutesMock as AppRoutes } from "./AppRoutes.mock";
