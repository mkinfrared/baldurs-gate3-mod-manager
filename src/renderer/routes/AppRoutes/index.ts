export { AppRoutes } from "./AppRoutes";

export * from "./AppRoutes.type";
