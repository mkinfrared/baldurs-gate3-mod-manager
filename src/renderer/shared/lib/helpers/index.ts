export { classNames } from "./classNames";

export { compose } from "./compose";

export { delay } from "./delay";

export { getNodeText } from "./getNodeText";

export { lazyImport } from "./lazyImport";

export { mergeRefs } from "./mergeRefs";

export * from "./typeGuards";

export { createTrpcClient, trpc } from "./trpc";
