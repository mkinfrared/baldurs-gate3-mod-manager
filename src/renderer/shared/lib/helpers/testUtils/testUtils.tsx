/* eslint-disable import/no-unused-modules */
import { userEvent } from "@storybook/testing-library";
import { render } from "@testing-library/react";
import { ReactElement } from "react";
import { BrowserRouter } from "react-router-dom";

// https://testing-library.com/docs/example-react-router/
const renderWithRouter = (ui: ReactElement, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

export { renderWithRouter };
