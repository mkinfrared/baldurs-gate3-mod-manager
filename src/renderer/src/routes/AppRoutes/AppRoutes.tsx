import { Navigate, Route, Routes } from "react-router-dom";

import { BaldursGate3, Settings } from "@renderer/pages";

import { APP_ROUTES } from "./AppRoutes.type";

const AppRoutes = () => (
  <Routes>
    <Route path={APP_ROUTES.BaldursGate3.path} element={<BaldursGate3 />} />
    <Route
      path={APP_ROUTES.BaldursGate3.SETTINGS.path}
      element={<Settings game="BG3" />}
    />
    <Route path="*" element={<Navigate to={APP_ROUTES.BaldursGate3.path} />} />
  </Routes>
);

export { AppRoutes };
