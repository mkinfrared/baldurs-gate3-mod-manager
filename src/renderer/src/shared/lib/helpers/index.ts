export { classNames } from "./classNames";

export { compose } from "./compose";

export { delay } from "./delay";

export { getNodeText } from "./getNodeText";

export { lazyImport } from "./lazyImport";

export { mergeRefs } from "./mergeRefs";

export { trpcClient, trpc } from "./trpc";

export * from "./typeGuards";
