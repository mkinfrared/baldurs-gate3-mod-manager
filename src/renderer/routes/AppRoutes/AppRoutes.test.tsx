import { renderWithRouter } from "renderer/shared/lib/helpers/testUtils";

import { AppRoutes } from "./AppRoutes";

describe("<AppRoutes />", () => {
  const Component = <AppRoutes />;

  it("should be defined", () => {
    expect(AppRoutes).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = renderWithRouter(Component);

    expect(container).toMatchSnapshot();
  });
});
