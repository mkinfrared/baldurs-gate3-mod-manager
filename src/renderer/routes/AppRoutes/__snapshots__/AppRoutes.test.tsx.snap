// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRoutes /> should match the snapshot 1`] = `
<div>
  <div
    class="BaldursGate3"
    data-testid="BaldursGate3"
  >
    <div
      class="InstalledModsSection"
      data-testid="InstalledModsSection"
    >
      <h3
        class="Heading"
        data-testid="Heading"
      >
        Installed Mods
      </h3>
      <div
        class="InstalledModsList"
        data-testid="InstalledModsList"
      >
        <table
          class="Table DataTable table"
          data-testid="Table"
        >
          <thead>
            <tr>
              <th>
                Active
              </th>
              <th>
                Name
              </th>
              <th>
                Version
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
    <div
      class="ModFileSection"
      data-testid="ModFileSection"
    >
      <h3
        class="Heading"
        data-testid="Heading"
      >
        Modification Files
      </h3>
      <div
        class="ModFileList"
        data-testid="ModFileList"
      />
    </div>
  </div>
</div>
`;
